src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -166,7 +166,9 @@ private static ControlFlowGraph GetControlFlowGraphCore(IOperation operation, st
 
             try
             {
-                return ControlFlowGraphBuilder.Create(operation);
+                ControlFlowGraph controlFlowGraph = ControlFlowGraphBuilder.Create(operation);
+                Debug.Assert(controlFlowGraph.OriginalOperation == operation);
```

Comment:
It feels like this assertion could be in `ControlFlowGraphBuilder.Create` itself.

Reply:
> It feels like this assertion could be in ControlFlowGraphBuilder.Create itself.

I prefer the observer to be outside.


---
In reply to: [193824693](https://github.com/dotnet/roslyn/pull/27536#discussion_r193824693) [](ancestors = 193824693)